.github/workflows/build-and-push.yaml

```diff
@@ -0,0 +1,104 @@
+    steps:
+    - uses: actions/checkout@v3
+    - uses: dorny/paths-filter@v2
+      id: changes
+      with:
+        filters: |
+          ansible-eda-base-image:
+            - 'ansible-eda-base-image/**'
+          ansible-eda:
+            - 'ansible-eda/**'
+
+  build-base-image:
+    name: Build EDA Base Image
+    runs-on: ubuntu-latest
+    needs: changes
+
+    # Only build the base image if changes are detected,
+    # or if this build was manually triggered
+    if: ${{ (needs.changes.outputs.ansible-eda-base-image == 'true') || (github.event_name == 'workflow_dispatch') }}
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Buildah Action
+      uses: redhat-actions/buildah-build@v2
+      id: build-image
+      with:
+        image: ansible-eda-base
+        tags: latest ${{ github.sha }}
+        platforms: linux/amd64
+        context: ./ansible-eda-base-image
+        containerfiles: ./ansible-eda-base-image/Containerfile
+
+    - name: Echo Build Outputs
+      run: |
+        echo "Image: ${{ steps.build-image.outputs.image }}"
+        echo "Tags: ${{ steps.build-image.outputs.tags }}"
+        echo "Tagged Image: ${{ steps.build-image.outputs.image-with-tag }}"
+
+    - name: Push To quay.io
+      id: push-to-quay
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: ${{ steps.build-image.outputs.image }}
+        tags: ${{ steps.build-image.outputs.tags }}
+        registry: quay.io/rhdevelopers
+        username: ${{ secrets.QUAY_USERNAME }}
+        password: ${{ secrets.QUAY_PASSWORD }}
+
+    - name: Print image url
+      run: echo "Image pushed to ${{ steps.push-to-quay.outputs.registry-paths }}"
+
+
+  build-eda-image:
+    name: Build EDA k8s Image
+    runs-on: ubuntu-latest
+    needs: build-base-image
+
+    # Build this image if changes are detected in this or the base image files
+    # or if this build was manually triggered
+    if: ${{ (needs.changes.outputs.ansible-eda-base-image == 'true') || (needs.changes.outputs.ansible-eda == 'true') || (github.event_name == 'workflow_dispatch') }}
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Buildah Action
+      uses: redhat-actions/buildah-build@v2
+      id: build-image
+      with:
+        image: k8s-ansible-eda
+        tags: mail
+        platforms: linux/amd64
+        context: ./ansible-eda
+        containerfiles: ./ansible-eda/Dockerfile
+
+    - name: Echo Build Outputs
+      run: |
+        echo "Image: ${{ steps.build-image.outputs.image }}"
+        echo "Tags: ${{ steps.build-image.outputs.tags }}"
+        echo "Tagged Image: ${{ steps.build-image.outputs.image-with-tag }}"
+
+    - name: Push To quay.io
+      id: push-to-quay
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: ${{ steps.build-image.outputs.image }}
+        tags: ${{ steps.build-image.outputs.tags }}
+        registry: quay.io/rhdevelopers
+        username: ${{ secrets.QUAY_USERNAME }}
```

Comment:
I have configured these variables on this repository, so once this PR is merged it will "just work" 😄

Reply:
Great